shared/qt/idian/include/Idian/Group.hpp

```diff
@@ -0,0 +1,75 @@
+/******************************************************************************
+    Copyright (C) 2023 by Dennis Sädtler <dennis@obsproject.com>
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+******************************************************************************/
+
+#pragma once
+
+#include <QLayout>
+#include <QLabel>
+#include <QWidget>
+#include <QMouseEvent>
+
+#include <Idian/PropertiesList.hpp>
+#include <Idian/Row.hpp>
+#include <Idian/ToggleSwitch.hpp>
+
+namespace idian {
+
+class Group : public QFrame, public Utils {
+	Q_OBJECT
+
+public:
+	Group(QWidget *parent = nullptr);
+
+	PropertiesList *properties() const { return propertyList; }
+
+	// ToDo add event for checkable group being enabled/disabled
+	// ToDo allow setting enabled state
+	// (Maybe) ToDo add option for hiding properties list when disabled
```

Comment:
See above regarding `TODO` blocks, also remove if this is not a necessary to-do, but actually just ideas for future features.